File: app/src/main/java/com/luoyang/myandroidxstudy/ui/MeViewModel.java
package com.luoyang.myandroidxstudy.ui;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.luoyang.myandroidxstudy.bean.User;
import com.luoyang.myandroidxstudy.net.NetServerEngine;
import com.luoyang.myandroidxstudy.util.MyCallback;

/**
 * @author LuoYang
 */
public class MeViewModel extends ViewModel {
    private MutableLiveData<User> data;

    public MutableLiveData<User> getData() {
        if (data == null) {
            data = new MutableLiveData<>();
        }
        NetServerEngine.getServer().getMe().enqueue(new MyCallback<>(data));
        return data;
    }
}

File: app/src/main/java/com/luoyang/myandroidxstudy/ui/MeViewModel.kt
package com.luoyang.myandroidxstudy.ui

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.luoyang.myandroidxstudy.bean.User
import com.luoyang.myandroidxstudy.net.NetServerEngine
import com.luoyang.myandroidxstudy.util.MyCallback

class MeViewModel : ViewModel() {
    var data = MutableLiveData<User>()
    fun meInfo() {
        NetServerEngine.instance.mNetServer.getMe().enqueue(MyCallback(data))
    }
}
